### PR TITLE
[Feature] Radius Graph

### PR DESCRIPTION
**Pull request by ksadowski13**
## Description
Function and module for creating the radius graph from the point set. Implemented in vanilla PyTorch by constructing NxN pairwise distance matrix. It's fully vectorized, but can consume a lot of memory for huge graphs, so it's more suited for molecules, proteins, small point clouds etc.

Functionalities:
- possibility to include self-loops
- possibility to change power of Minkowski metric
- optionally returns distances for the constructed edges
- returns graph on the same device as point set

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by BarclayII** [comment]
@mufeili Where are we putting the graph construction modules?  Should they belong to `dgl.transform` or `dgl.nn.pytorch`?

**Review by hetong007 — APPROVED** [review]
Thanks, in general LGTM! Also, can we have a test for the new module?

**Comment by mufeili** [comment]
> @mufeili Where are we putting the graph construction modules? Should they belong to `dgl.transform` or `dgl.nn.pytorch`?

We can put it in `dgl/transforms/functional.py` as `knn_graph`.

**Comment by mufeili** [comment]
@ksadowski13 Could you update the doc indexing file [here](https://github.com/dmlc/dgl/blob/master/docs/source/api/python/dgl.rst)?

**Comment by ksadowski13** [comment]
@mufeili @hetong007 Sure, I will add changes at the beginning of the next week 😉

File: python/dgl/transforms/functional.py [hunk 1]
```diff
@@ -3300,4 +3302,108 @@ def norm_by_dst(g, etype=None):
...
+    """
+    # check invalid r
+    if r <= 0:
+        raise DGLError("Invalid r value. expect r > 0, got r = {}".format(r))
+
+    # check empty point set
+    if F.shape(x)[0] == 0:
+        raise DGLError("Find empty point set")
+
+    distances = th.cdist(x, x, p=p, compute_mode='donot_use_mm_for_euclid_dist')
```
[the remark this answers is not in the document]
Comment (ksadowski13) [reply python/dgl/transforms/functional.py]: Done

**Comment by ksadowski13** [comment]
> Thanks, in general LGTM! Also, can we have a test for the new module?

Done

**Comment by ksadowski13** [comment]
> @ksadowski13 Could you update the doc indexing file [here](https://github.com/dmlc/dgl/blob/master/docs/source/api/python/dgl.rst)?

Done

**Comment by mufeili** [comment]
> > @ksadowski13 Could you update the doc indexing file [here](https://github.com/dmlc/dgl/blob/master/docs/source/api/python/dgl.rst)?
> 
> Done

Thanks. I'm good.

**Pull request merged**